cmd/libsnap-confine-private/device-cgroup-support.c

```diff
@@ -342,16 +336,21 @@ static int _sc_cgroup_v2_init_bpf(sc_device_cgroup *self, int flags) {
         debug("bpffs mounted at /sys/fs/bpf");
     }
 
-    if (mkdir("/sys/fs/bpf/snap", 0700) < 0) {
-        if (errno != EEXIST) {
-            die("cannot create /sys/fs/bpf/snap directory");
+    /* TODO: open("/sys/fs/bpf") and then mkdirat?  */
+    static const char bpf_snap_base[] = "/sys/fs/bpf/snap";
+    if (mkdir(bpf_snap_base, 0700) == 0) {
```

Comment:
```
$ sudo ls -ld /sys/fs/bpf
drwx-----T 2 root root 0 Jun 28 16:32 /sys/fs/bpf
```

Are we sure that subdirectories could even be created by processes that aren't running as root? Or are we going to rely upon `CAP_DAC_OVERRIDE` so that DAC permissions don't matter at all? Will we juggle the `CAP_DAC_OVERRIDE` in the effective capability permission set in much the same way that we juggle uids now?

Reply:
Yes, it's understood that snap-confine will continue to run with CAP_DAC_OVERRIDE and a few other capabilities in order to perform all the actions that the current snap-confine does.

For the time being I would be happy enough if we could just migrate from being running as root to having a minimal set of capabilities (permanently, without juggling); as step two, we would try to also limit the time when they are used.

Note that currently the juggling of IDs is just to change the group ID, to make sure that directories and files are created with GID zero; in other words, snap-confine is currently always running as root.

Reply:
Aha, so if it is effectively always running with the full set of capabilities today, then this is really a narrowing of privileges, not the partial-narrowing and partial-widening that I expected. Thanks